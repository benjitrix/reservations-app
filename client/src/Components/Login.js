import React, {useState, useContext} from 'react';
import AuthService from '../Services/AuthServices'
import { AuthContext } from '../Context/AuthContext'
import Message from './Message'

// user login component
const Login = (props) => {
  const [user, setUser] = useState({username: "", password: "", role: ""});
  const [message, setMessage] = useState(null);
  const authContext = useContext(AuthContext);

  // capture username and password from input boxes, add to array
  const onChange = (e) => {
    setUser({...user, [e.target.name] : e.target.value})
  }

  // Login for user registered user using username and password
  const onSubmit = (e) => {
    e.preventDefault();
    AuthService.login(user).then(data => {
      console.log(data)
      const { isAuthenticated, user } = data;
      if (isAuthenticated) {
        authContext.setUser(user);
        authContext.setIsAuthenticated(isAuthenticated);
        if (user.role === "user") {
          props.history.push('/reservations');
        } else {
          props.history.push("/admin");
        }
      } else {
        setMessage(message);
      }
    }) 
  }

  return (
    <div>
      <form onSubmit={onSubmit}>
      <h3>Please sign in</h3>
      <label htmlFor="username" className="sr-only">Username: </label>
      <input  type="text" 
              name="username" 
              onChange={onChange} 
              className="form-control" 
              placeholder="Enter username" />
      <input  type="password" 
              name="password" 
              onChange={onChange} 
              className="form-control" 
              placeholder="Enter password" />

      <button type="submit" className="btn btn-lg btn-primary btn-block">Log in</button>
      </form>
      {message ? <Message message={message} /> : null}
    </div>
  )
}

export default Login
