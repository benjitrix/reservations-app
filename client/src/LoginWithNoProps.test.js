import React from 'react';
import Login from './Components/Login'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

// Testing Login component ui and props (no props passed) with snapshot and find selector
describe('<Login /> with no props', () => {
  const container = shallow(<Login />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have a username field', () => {
    expect(container.find('input[type="text"]').length).toEqual(1);
  });

  it('should have proper props for username field', () => {
    expect(container.find('input[type="text"]').props()).toEqual({
      className: "form-control",
      name: "username",
      onChange: expect.any(Function),
      placeholder: "Enter username",
      type: "text",
    });
  });

  it('should have a password field', () => {
    expect(container.find('input[type="password"]').length).toEqual(1);
  });

  it('should have proper props for password field', () => {
    expect(container.find('input[type="password"]').props()).toEqual({
      className: "form-control",
      name: "password",
      onChange: expect.any(Function),
      placeholder: "Enter password",
      type: "password",
    });
  });

});
