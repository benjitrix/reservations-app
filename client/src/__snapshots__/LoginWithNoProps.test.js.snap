// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props <Login /> with no props should match the snapshot 1`] = `"<div><form><h3>Please sign in</h3><label for=\\"username\\" class=\\"sr-only\\">Username: </label><input type=\\"text\\" name=\\"username\\" class=\\"form-control\\" placeholder=\\"Enter username\\"/><input type=\\"password\\" name=\\"password\\" class=\\"form-control\\" placeholder=\\"Enter password\\"/><button type=\\"submit\\" class=\\"btn btn-lg btn-primary btn-block\\">Log in</button></form></div>"`;

exports[`<Login /> with no props should match the snapshot 1`] = `"<div><form><h3>Please sign in</h3><label for=\\"username\\" class=\\"sr-only\\">Username: </label><input type=\\"text\\" name=\\"username\\" class=\\"form-control\\" placeholder=\\"Enter username\\"/><input type=\\"password\\" name=\\"password\\" class=\\"form-control\\" placeholder=\\"Enter password\\"/><button type=\\"submit\\" class=\\"btn btn-lg btn-primary btn-block\\">Log in</button></form></div>"`;
